File: Problem Set 6/pset6/readability/readability.py
from cs50 import get_string

t = get_string("text: ")

l = s = w = 0

for char in t:
    if char.isalpha():
        l += 1
    if char.isspace():
        w += 1
    if char in ["?", ".", "!"]:
        s += 1
        
w += 1
L = round(((l * 100.0) // w), 2)
S = round(((s * 100.0) // w), 2)
result = round(0.0588 * L - 0.296 * S - 15.8)
if result < 1:
    print("Before Grade 1")
elif result >= 16:
    print("Grade 16+")
else:
    print(f"Grade {result}")

File: Problem Set 2/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int letter_count(string text);
int word_count(string text);
int sentence_count(string text);

int main(void)
{
    string text = get_string("Text: ");
    printf("%s\n", text);
    double L, S;
    int index;
    int letter = letter_count(text);
    int word = word_count(text);
    int sen = sentence_count(text);
    L = (letter * 100.0f) / word;
    S = (sen * 100.0f) / word;
    index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int letter_count(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        char c = text[i];
        if (isalpha(c))
        {
            count++;
        }
    }
    return count;
}

int word_count(string text)
{
    int count = 1;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        char c = text[i];
        if (isspace(c))
        {
            count++;
        }
    }
    return count;
}

int sentence_count(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        char c = text[i];
        if (c == '.')
        {
            count++;
        }
        else if (c == '!')
        {
            count++;
        }
        else if (c == '?')
        {
            count++;
        }
    }
    return count;
}